lib/tlscontext.c

```diff
@@ -61,6 +64,10 @@ struct _TLSContext
   gchar *location;
 };
 
+#define _TLS_KEYLOG_INDEX 0
+#define _TLS_KEYLOG_FILE 1
+#define _TLS_KEYLOG_FILE_LOCK 2
```

Comment:
It turned out that `SSL_get_ex_data()` is not really necessary, because we already use `SSL_get_app_data()`.
This way, we don't need to worry about OpenSSL versions either.

For example:
```C
TLSSession *self = SSL_get_app_data(ssl);
self->ctx->keylog_file
self->ctx->keylog_file_path
self->ctx->keylog_file_lock
```